src/perf/event_provider.cpp

```diff
@@ -687,23 +350,109 @@ bool EventProvider::has_event(const std::string& name)
     }
 }
 
-std::vector<EventDescription> EventProvider::get_predefined_events()
+std::vector<Event> EventProvider::get_predefined_events()
 {
-
     const auto& ev_map = instance().event_map_;
 
-    std::vector<EventDescription> events;
+    std::vector<Event> events;
     events.reserve(ev_map.size());
 
     for (const auto& event : ev_map)
     {
         if (event.second.is_valid())
         {
-            events.push_back(event.second.description);
+            events.push_back(std::move(event.second));
         }
     }
 
     return events;
 }
+
+tracepoint::TracepointEvent EventProvider::create_tracepoint_event(const std::string& name,
+                                                                   const bool& enable_on_exec)
+{
+    tracepoint::TracepointEvent event(name, enable_on_exec);
+    event.set_sample_period(0);
+    apply_config_attrs(event);
+
+    return event;
+}
+
+tracepoint::TracepointEvent EventProvider::create_raw_tracepoint_event(const std::string& name,
+                                                                       const bool& enable_on_exec)
+{
+    tracepoint::TracepointEvent event(name, enable_on_exec);
+    event.set_sample_period(0);
+    apply_default_attrs(event);
+
+    return event;
+}
```

Comment:
these two functions are identical, but the name.

Reply:
They are not, as one uses config() and the other does not. 
But I agree, they are pretty similar. Will merge them into one.